### PR TITLE
right click menu has been disabled, right click has been freed

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Ports this from is12, thanks matt. All credit to him.

Admins still have the ability to setup the pop up menu.

Safety is now right click for guns as a proof of concept

## Why It's Good For The Game
GIves us way more control over our controls. And allows for more use of the mouse rather than having an antiquated shhitty right click menu

## Changelog
:cl:
del: Right click menu is now toggleable in preferences
tweak: moved safety from alt click to right click
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -145. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by BeloneX** [comment]
What the fuck, why?

**Comment by Chaznoodles** [comment]
> What the fuck, why?
> 
> Nevermind the countless things that require the right click for various functionalities, this is straight-up cursed.

You can alt-click rather than right-click to view stacks and pick things up, it also tends to be less laggy on the client. That's pretty much all right-click is used for.

**Comment by BeloneX** [comment]
> > What the fuck, why?
> > Nevermind the countless things that require the right click for various functionalities, this is straight-up cursed.
> 
> You can alt-click rather than right-click to view stacks and pick things up, it also tends to be less laggy on the client. That's pretty much all right-click is used for.

Countless things require the right-click menu to toggle specific settings or functionalities.

**Comment by Pariah919** [comment]
>What the fuck, why?

>Nevermind the countless things that require the right click for various functionalities, this is straight-up cursed.

Move them to alt right click, right click, alt right click.. etc.. 
Right click menu is a curse on ss13, removing it gives us vastly more control  over the controls

**Comment by BeloneX** [comment]
Just saw this is a preference, nevermind then.

You should make it be right-clicked by default or else this weird out a lot of people like when Xeno abilities were set to be shift-click by default.

**Review by BeloneX — COMMENTED** [review]
Misleading changelog, mention that it's a preference you can toggle instead.

**Comment by Pariah919** [comment]
>You should make it be right-clicked by default or else this weird out a lot of people like when Xeno abilities were set to be shift-click by default.

right click not being tied to the menu should become the default, and the right click menu should be an exception

**Comment by Tennessee116** [comment]
> Move them to alt right click, right click, alt right click.. etc..

If you aren't going to do that, don't merge this.

**Comment by davidchan13** [comment]
Right click works from more than 1 tile away. Alt Clicking requires you to be next to the tile to it.

This makes checking for hugger holes, stealthing/hiding xenos and larva, mines, obscured acid wells and any number of things you'd want to do safely at a distance assanine to outright impossible.

**Comment by SplinterGP** [comment]
> You can alt-click rather than right-click to view stacks and pick things up, it also tends to be less laggy on the client. That's pretty much all right-click is used for.

that's actually not true, you cant use alt-click on distance and alt-click is way slower than right clicking as right click is just a windows thingy and the alt click window is slow as it doenst has an HTML stat-panel, making it way more laggy than right clicking, also right click has "move-to-top" which makes the item be on the top of the piles, nothing much  but helps on organization.

**Comment by TiviPlus** [comment]
@BeloneX Do not misuse the review feature

File: code/_onclick/click.dm [hunk 1]
```diff
@@ -85,6 +85,15 @@
...
 		return
+	if(modifiers["shift"] && modifiers["right"])
+		ShiftRightClickOn(A)
+		return 1
+	if(modifiers["alt"] && modifiers["right"])
+		AltRightClickOn(A)
+		return 1
+	if(modifiers["right"])
+		RightClickOn(A)
+		return
```
Comment (TiviPlus) [inline code/_onclick/click.dm:96]:
```suggestion
	if(modifiers["shift"] && modifiers["right"])
		ShiftRightClickOn(A)
		return
	if(modifiers["alt"] && modifiers["right"])
		return
	if(modifiers["right"] && RightClickOn(A))
		return
```

File: code/modules/client/preferences_toggles.dm [hunk 2]
```diff
@@ -330,3 +330,14 @@ GLOBAL_LIST_INIT(ghost_others_options, list(GHOST_OTHERS_SIMPLE, GHOST_OTHERS_DE
...
+client/verb/toggle_right_click()
+	set name = "Toggle Right Click"
+	set category = "Preferences"
+
+	if(!show_popup_menus)
+		show_popup_menus = TRUE
+		to_chat(src, "<span class='interface'>Right click enabled.</span>")
+	else
+		show_popup_menus = FALSE
+		to_chat(src, "<span class='interface'>Right click disabled.</span>")
```
Comment (TiviPlus) [inline code/modules/client/preferences_toggles.dm:343]:
```suggestion
	show_popup_menus = !show_popup_menus
	to_chat(src, "<span class='interface'>Right click [show_popup_menus ? "en" : "dis"]abled.</span>")
```

File: code/_onclick/click.dm [hunk 3]
```diff
@@ -319,6 +328,29 @@ if(selected_ability.target_flags & flagname){\
...
+/mob/proc/AltRightClickOn(atom/A)
+	A.AltRightClick(src)
+
+/atom/proc/ShiftRightClick(mob/user)
+	return
+
+/atom/proc/AltRightClick(mob/user)
+	return
+
+/atom/proc/RightClick(mob/user)
```
Comment (TiviPlus) [inline code/_onclick/click.dm:351]:
```suggestion

///Called when a owner mob Rightmouseclicks an atom
/mob/proc/RightClickOn(atom/A)
	A.RightClick(src)

///Called when a owner mob Shift + Rightmouseclicks an atom
/mob/proc/ShiftRightClickOn(atom/A)
	A.ShiftRightClick(src)

///Called when a owner mob Alt + Rightmouseclicks an atom, given that Altclick does not return TRUE
/mob/proc/AltRightClickOn(atom/A)
	A.AltRightClick(src)

///Called when a mob Rightmouseclicks this atom
/atom/proc/RightClick(mob/user)
	return

///Called when a mob Shift + Rightmouseclicks this atom
/atom/proc/ShiftRightClick(mob/user)
	return
	
///Called when a mob Alt + Rightmouseclicks this atom, given that mobs Altclick() does not return TRUE
/atom/proc/AltRightClick(mob/user)
```

File: code/modules/client/preferences_toggles.dm [hunk 4]
```diff
@@ -330,3 +330,14 @@ GLOBAL_LIST_INIT(ghost_others_options, list(GHOST_OTHERS_SIMPLE, GHOST_OTHERS_DE
 	usr.client.prefs.save_preferences()
 
 	to_chat(usr, "<span class='notice'>You will [(usr.client.prefs.toggles_sound & SOUND_INSTRUMENTS_OFF) ? "no longer" : "now"] hear instruments.</span>")
+
+client/verb/toggle_right_click()
```
Comment (TiviPlus) [inline code/modules/client/preferences_toggles.dm:334]:
```suggestion

///Toggles the right click menu on and off
/client/verb/toggle_right_click()
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by BraveMole** [comment]
I suppose that loading sprites into the alt click menu is making things laggy, maybe make it a preference so that people having issue with that menu can use it more

**Comment by SpaceLoveSs13** [comment]
Will Alt-click let us use the "Change pinpointer" on Advanced pinpointers to direct user to a Nuclear Fission Device or on going disk generator? Also right click change to tac binos [I know shift-click exists for tacbino modes, but right click shows new people the way to do it at first]

**Comment by fernandos13** [comment]
make it preference
and make so people can CHOOSE whatever they bind to it. safety for me is almost useless tbh. but have it set to pick up health analyzer macro might be kinda better than examine tiles....

**Review by TiviPlus — APPROVED** [review]

**Comment by TiviPlus** [comment]
Im gonna hold off on merging this till either myself or someone else is available to make more rightclick functions

**Review by SplinterGP — APPROVED** [review]
fine for me, should keep a way to use right click with it off, be it ctrl right click or something, some stuff only has verbs which can be either used on top right tabs or shows on right click.

**Comment by Pariah919** [comment]
point to bothering

**Comment by Pariah919** [comment]
there's nothing of value left here

**Comment by Pariah919** [comment]
i might as well leave this here in hindsight rather than delete it

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-257). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**